web_monitoring/db.py

```diff
@@ -45,3 +47,89 @@ def query_import_status(import_id):
                                                 import_id=import_id)
     return requests.get(url,
                         auth=(settings['db_email'], settings['db_password']))
+
+def get_version_uri(version_id, id_type='db', source_type='versionista', get_previous=False):
+    """
+    Get the uri of a version(snapshot) stored in the db.
+
+    Parameters
+    ----------
+    version_id* : string
+    source_type : string
+    get_previous : boolean
+    * -> required fields
+    """
+    if (id_type == 'db'):
+        url = WEB_MONITORING_GET_VERSION_API.format(db_url=settings['db_url'],
+                                                    version_id=version_id)
+    elif (id_type == 'source'):
+        url = WEB_MONITORING_GET_VERSION_SOURCE_API.format(db_url=settings['db_url'],
+                                                           source_type=source_type,
+                                                           version_id=version_id)
+    else:
+        raise ValueError('Id type should be either "db" or "source"')
+
+    response = requests.get(url,
+                            auth=(settings['db_email'], settings['db_password']))
+    result = response.json()
+
+    if (len(result['data']) == 0):
+        if(get_previous):
```

Comment:
I was not too sure about what to do here. Although it does make sense to raise an Exception here for a consumer trying to get a single version uri, it causes problems when a script like `data_downloader` tries to get a big list of uris. Since the downloading script depends on the data in the csv, anyone using it will have to make sure that the data is clean and the versions exist in the db (which is a little tedious).

Reply:
A good point. I would say that the right place to swallow the error is in `data_downloader`. When it gets an exception, it can decide to (1) silently ignore it, (2) print a warning, (3) return a list of the uudis of the failures, or (4) blow up. By passing up a warnings (instead of `None`), you will be providing `data_downloader` with more detailed information about the _reason_ for the failure.

Reply:
Right, warnings will be more helpful than just a `None`. I'll print warnings and get a list of failed uuids as well (the logs should come in handy here).